src/blocks.zig

```diff
@@ -414,39 +418,108 @@ pub const Block = packed struct { // MARK: Block
 };
 
 // MARK: Tick
-pub var tickFunctions: utils.NamedCallbacks(TickFunctions, TickFunction) = undefined;
-pub const TickFunction = fn(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void;
-pub const TickFunctions = struct {
-	pub fn replaceWithCobble(block: Block, _chunk: *chunk.ServerChunk, x: i32, y: i32, z: i32) void {
-		std.log.debug("Replace with cobblestone at ({d},{d},{d})", .{x, y, z});
-		const cobblestone = parseBlock("cubyz:cobblestone");
+pub const TickEventVTableMap = struct {
+	const Self = @This();
 
-		const wx = _chunk.super.pos.wx + x;
-		const wy = _chunk.super.pos.wy + y;
-		const wz = _chunk.super.pos.wz + z;
+	vTableMap: std.StringHashMap(*const TickEvent.VTable) = undefined,
+
+	pub fn init() Self {
+		var self = Self{
+			.vTableMap = .init(main.globalAllocator.allocator),
+		};
+
+		switch(@typeInfo(TickEvents)) {
+			.@"struct" => |eventStruct| {
```

Comment:
Please try to avoid such deep indentations. We don't need a switch, where an if is good enough.
Also this should be a compiler error and not a runtime error.
```
const typeInfo = @typeInfo(TickEvents);
if(typeInfo != .struct) retturn @compilerError("Type " ++ @typeName(TickEvents) ++ "is not a struct");

inline for(typeInfo.struct.decls) ...
```